persistence-modules/hibernate6/src/main/java/com/baeldung/attributevaluegenerator/SortingHatHouseGenerator.java

```diff
@@ -0,0 +1,27 @@
+package com.baeldung.attributevaluegenerator;
+
+import java.util.EnumSet;
+import java.util.Random;
+
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.generator.BeforeExecutionGenerator;
+import org.hibernate.generator.EventType;
+
+public class SortingHatHouseGenerator implements BeforeExecutionGenerator {
+
+    private static final String[] HOUSES = { "Gryffindor", "Hufflepuff", "Ravenclaw", "Slytherin" };
+    private static final Random RANDOM = new Random();
```

Comment:
It doesn't matter that much but you should generally never use Random any more, always use ThreadLocalRandom
```suggestion
    private static final Random RANDOM = new ThreadLocalRandom();
```

Reply:
got it, thanks. Incorporated.